### PR TITLE
Robustify Rayleigh Bernard two phase test

### PR DESCRIPTION
**Pull request by lindsayad**
Refs failures from the combination of #26462 and #26227

To avoid negative viscosity interpolations I shrank the growth factor from 2 to 1.5. Once I did this I got a CSVDiff which signaled to me that the original result was solved to too loose a tolerance. So I removed the steady state detection, which has potential issues with it (see #26312), and extended the `end_time` until we hit a fairly tight absolute tolerance

### CODE REVIEW

**Comment by lindsayad** [comment]
FYI @loganharbour @tanoret

**Comment by lindsayad** [comment]
the diff is quite large; it makes me concerned there is something genuinely wrong either with distributed mesh or with threads

**Comment by tanoret** [comment]
> the diff is quite large; it makes me concerned there is something genuinely wrong either with distributed mesh or with threads

This is mostly because the problem admits multiple solutions, which makes it a bit sensitive to the initial seed in the solution. We can always invert gravity and make the flow stagnate to avoid multiple solutions, but then the problem is not very interesting anymore. What do you think?

**Comment by lindsayad** [comment]
I think that if the test is driving to a steady-state, the test should only have a single unique solution. We want the test to be deterministic; we don't want the result to be influenced by small perturbations in the order of floating point arithmetic. Or if the test input admits multiple solutions, then we need to modify what we are testing

**Comment by tanoret** [comment]
Actually, in steady-state the test admits multiple solutions; rising plume by the center of the domain and down by the sides, rising plume by the sides of the domain and down by the center, and these permutations exists for all the eddies formed (e.g., https://journals.aps.org/pre/abstract/10.1103/PhysRevE.105.055106). Most natural-convection-driven problems for two-phase flow will admit multiple solutions. That is why I think we need a test with a high tolerance to encompass to guarantee no regression  in the multiple solutions that may exist in the problem. If you guys don't like this, we can always invert gravity, stratify the flow, and go into a unique solution case. I am good with any of these cases

**Comment by lindsayad** [comment]
> That is why I think we need a test with a high tolerance

I don't really understand what this means. This sounds like a problem that has just been solved incompletely. A problem solved incompletely seems like it could be a solve that is on its way to solution 1, or to solution 2, or since it's solved incompletely a superposition of them. How can you be certain that the solver will repeatably go to the same incompletely solved solution? As soon as we updated PETSc we were perturbed to a different state, which is why next is red https://civet.inl.gov/branch/6695/. So the loose tolerance from before was not leading to robust, repeatable final test results

**Comment by lindsayad** [comment]
the results in this test are very cool by the way. very cool new functionality. now we just have to figure out what to do with civet

**Comment by tanoret** [comment]
> > a test with a high tolerance
> 
> I think I may have interpreted this wrong. I assumed you meant a loose solve tolerance, but perhaps you meant a loose CSVDiff tolerance? That does make more sense to me

Yes. That is the one, loose CSV tolerance. Should we make it looser? Is that possible? If we break something up, it should still show up if the tolerance is small enough

**Comment by lindsayad** [comment]
see my newest commit which just changes nx and ny to 11 instead of 10. This seems to lead to these repeatable solutions for vel-x and vel-y
Screenshot 2024-01-29 at 1 40 42 PM
Screenshot 2024-01-29 at 1 41 16 PM

**Comment by tanoret** [comment]
Yes. Looks great! Nice idea introducing anisotropy in the mesh

File: modules/navier_stokes/src/functormaterials/WCNSFV2PSlipVelocityFunctorMaterial.C [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  params.addRequiredParam<MooseFunctorName>("rho_d", "Dispersed phase density.");
+  params.addRequiredParam<MooseFunctorName>(NS::mu, "Mixture Density");
+  params.addParam<RealVectorValue>(
+      "gravity", RealVectorValue(0, 0, 0), "Gravity acceleration vector");
+  params.addParam<Real>("force_value", 0.0, "Coefficient to multiply by the body force term");
+  params.addParam<FunctionName>("force_function", "0", "A function that describes the body force");
+  params.addParam<PostprocessorName>(
+      "force_postprocessor", 0, "A postprocessor whose value is multiplied by the body force");
+  params.addParam<RealVectorValue>(
+      "force_direction", RealVectorValue(1, 0, 0), "Gravity acceleration vector");
```
Comment (grmnptr) [inline modules/navier_stokes/src/functormaterials/WCNSFV2PSlipVelocityFunctorMaterial.C:35]:
```suggestion
      "force_direction", RealVectorValue(1, 0, 0), "Gravitational acceleration vector");
```

**Review by grmnptr — APPROVED** [review]
Looks good to me!

File: modules/navier_stokes/src/functormaterials/WCNSFV2PSlipVelocityFunctorMaterial.C (hunk 1 above)
Comment (lindsayad) [reply modules/navier_stokes/src/functormaterials/WCNSFV2PSlipVelocityFunctorMaterial.C:35]: I'll apply this in #26683

**Comment by lindsayad** [comment]
failures are unrelated

File: modules/navier_stokes/src/functormaterials/WCNSFV2PSlipVelocityFunctorMaterial.C (hunk 1 above)
Comment (grmnptr) [reply modules/navier_stokes/src/functormaterials/WCNSFV2PSlipVelocityFunctorMaterial.C:35]: Sounds good!

**Pull request merged**